http-service/src/main/java/net/runelite/http/service/chat/ChatController.java

```diff
@@ -250,9 +252,12 @@ public void submitLayout(@RequestParam String name, @RequestBody LayoutRoom[] ro
 	}
 
 	@PostMapping("/pets")
-	public void submitPetList(@RequestParam String name, @RequestBody int[] petList)
+	public void submitPetList(@RequestParam String name, @RequestBody HashMap<Integer, Integer> petList)
 	{
-		if (petList.length == 0 || petList.length > MAX_PETS)
+		if (petList.size() == 0
+			|| petList.size() > MAX_PETS
+			|| Collections.max(petList.values()) > MAX_PETS
```

Comment:
Huh? Why would it matter if a user has received a duplicate pet more than 256 times?

Reply:
This was for data validation. As the Collection Log only counts a given entry up to 250 times, any amount above 250 would definitely be an error state. Can change `MAX_PETS` to 250

Reply:
Is this confirmed somewhere? There are numerous log entries which definitely count beyond 256 of a drop, though I believe each entry can have a different maximum counted value.